bin/send_email.py

```diff
@@ -6,60 +6,152 @@
+
+        Arguments:
+          status: The submission status
+          email_with_score: '0' if e-mail should not include score value / link to submissions views
+          submission_id: The submission ID of the given submission on Synapse
+          view_id: The Submission view ID on Synapse
+          score: The score value of the submission
+          reason: The reason for the validation error, if present.
+
+        Returns:
+          A string for that represents the body of the e-mail to be sent out to submitting team or individual.
+
+        """
+        templates = {
+            (
+                "VALIDATED",
+                "1",
+            ): f"Submission {submission_id} has been evaluated with a score value of {str(score)}. View all your scores here: https://www.synapse.org/#!Synapse:{view_id}/tables/",
+            (
+                "VALIDATED",
+                "0",
+            ): f"Submission {submission_id} has been evaluated. Your score will be available after Challenge submissions are closed. Thank you for participating!",
+            (
+                "INVALID",
+                "1",
+            ): f"Evaluation failed for Submission {submission_id}."
+            + "\n"
+            + f"Reason: '{reason}'."
+            + "\n"
+            + f"View your submissions here: https://www.synapse.org/#!Synapse:{view_id}/tables/, and contact the organizers for more information.",
+            (
+                "INVALID",
+                "0",
+            ): f"Evaluation failed for Submission {submission_id}."
+            + "\n"
+            + f"Reason: '{reason}'."
+            + "\n"
+            + "Please contact the organizers for more information.",
+        }
+
+        body = templates.get((status, email_with_score))
+
+        return body
+
+    def get_annotations(self, submission_id: str):
+        """
+        Gets the ``status`` ``score`` and ``reason`` annotations for the given
+        submission on Synapse and makes them attributes of the class instance.
+
+        1. ``status`` is the submission status, as defined by the last begun stage
+        in the MODEL_TO_SYNAPSE workflow.
+        2. ``score`` is the score of the model, used to determine its accuracy.
+        3. ``reason`` is the reason for the validation error, if there was one.
+        It remains an empty string (None) if no validation error.
+        """
+        submission_annotations = self.syn.getSubmissionStatus(submission_id)[
+            "submissionAnnotations"
+        ]
+        # TODO: "auc" may not always be the annotation name for score.
+        # Should enforce annotation names in the score/validation scripts.
+        self.status = submission_annotations.get("validation_status")[0]
+        self.score = submission_annotations.get("auc")[0]
+        self.reason = submission_annotations.get("validation_errors")[0]
+
+    def send_email(self, view_id: str, submission_id: str, email_with_score: str):
```

Comment:
nit: Return type

Is there any status indicator we can return for a success/failure?

Reply:
Yeah - I was thinking along the line of "If something went wrong, how would I know"

Reply:
I've implemented a few error catch outside of this thread that would reflect on the tower UI should things go wrong, but I think beyond that, the bin scripts would benefit from adding logging to ensure that e-mails are sent to the right people, for example. I think that would be a larger initiative that can go in another PR, but for now the error handling I have in place should be enough for this one.